liveblog.php

```diff
@@ -462,21 +498,39 @@ public static function ajax_crud_entry() {
 		$args['content'] = isset( $_POST['content'] ) ? $_POST['content'] : '';
 		$args['entry_id'] = isset( $_POST['entry_id'] ) ? intval( $_POST['entry_id'] ) : 0;
 
-		$args['user'] = wp_get_current_user();
-
-		$entry = call_user_func( array( 'WPCOM_Liveblog_Entry', $crud_action ), $args );
+		$entry = self::do_crud_entry($crud_action, $args);
 
 		if ( is_wp_error( $entry ) ) {
 			self::send_server_error( $entry->get_error_message() );
 		}
 
-		// Do not send latest_timestamp. If we send it the client won't get
-		// older entries. Since we send only the new one, we don't know if there
-		// weren't any entries in between.
-		self::json_return( array(
-			'entries'           => array( $entry->for_json() ),
-			'latest_timestamp'  => null
-		) );
+		self::json_return( $entry );
+	}
+
+	/**
+	 * Perform a specific CRUD action on an entry for a post
+	 *
+	 * @param string $crud_action Allowed actions are insert|update|delete|delete_key
+	 * @param array $args An array of data to be passed to the crud method
+	 *
+	 * @return mixed The result of the crud method
+	 */
+	public static function do_crud_entry( $crud_action, $args ) {
```

Comment:
I ended up putting the `crud_action` validation in it's own method called `is_valid_crud_action`. This way we can still take advantage of the built in validation callback functionality in the REST API.